lightning/cogs/automod/ui.py

```diff
@@ -191,3 +203,338 @@ async def warn_ban(self, itx: discord.Interaction, button: discord.ui.Button):
+                   f"**Verification Role**: {self.gatekeeper.role.mention}\n"\
+                   f"**Verification Channel**: {self.gatekeeper.verification_channel.mention}"
+            for button in setup_buttons:
+                button.disabled = True
+            self.set_switch_labels(True)
+            return text
+
+        self.send_verification_message.disabled = True
+
+        if record is None:
+            text = "Lightning Gatekeeper is not fully set up!"
+            self.set_gatekeeper_role.disabled = False
+            self.set_gatekeeper_channel.disabled = True
+            self.disable_gatekeeper.disabled = True
+            return text
+        elif record['role_id'] is None:
+            text = "Lightning Gatekeeper is not fully set up!"
+            self.disable_gatekeeper.disabled = True
+            self.set_gatekeeper_channel.disabled = True
+            self.set_switch_labels(False)
+            return text
+        elif record['verification_channel_id'] is None:
+            text = "Lightning Gatekeeper is not fully set up!"
+            self.set_gatekeeper_role.disabled = True
+            self.disable_gatekeeper.disabled = True
+            self.set_gatekeeper_channel.disabled = False
+            self.set_switch_labels(False)
+            return text
+        elif record['active'] is False:
+            text = "Lightning Gatekeeper is currently disabled!\n"\
+                   "*Click the Enable button to enable the gatekeeper for everyone*"
+            self.disable_gatekeeper.disabled = False
+            for button in setup_buttons:
+                button.disabled = False
+            self.set_switch_labels(False)
+
+        if record['verification_channel_id'] and record['role_id']:
+            self.disable_gatekeeper.disabled = False
+            self.send_verification_message.disabled = False
+
+        return text
+
+    def invalidate_gatekeeper_cache(self):
+        self.ctx.cog.invalidate_gatekeeper(self.ctx.guild.id)
+
+    def set_switch_labels(self, status: bool):
+        if status:
+            self.disable_gatekeeper.label = "Disable"
+            self.disable_gatekeeper.style = discord.ButtonStyle.red
+        else:
+            self.disable_gatekeeper.label = "Enable"
+            self.disable_gatekeeper.style = discord.ButtonStyle.green
+
+    @discord.ui.button(label="Set a gatekeeper role", style=discord.ButtonStyle.blurple)
+    async def set_gatekeeper_role(self, itx: discord.Interaction[LightningBot], button: discord.ui.Button):
+        view = GatekeeperRoleView(author_id=itx.user.id, clear_view_after=True)
+        await itx.response.send_message(content='Select a role from the select menu below or '
+                                        'create a new role by clicking the "Create a New Role" button',
+                                        view=view, ephemeral=True)
+        await view.wait()
+        self.invalidate_gatekeeper_cache()
+        await self.update(interaction=itx)
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (edge_case_not_handled):** Validate channel selection for gatekeeper setup.

Ensure that the selected channel for gatekeeper verification is appropriate and has the necessary permissions set before updating the configuration.

```suggestion
        if not channel.permissions_for(itx.guild.me).manage_channels:
            await itx.followup.send("I do not have permission to manage channels, which is necessary for setting up the verification channel.", ephemeral=True)
            return

        if not channel.permissions_for(itx.guild.default_role).read_messages:
            await itx.followup.send("The selected channel already restricts read access to everyone. Please select a channel where everyone can initially read messages.", ephemeral=True)
            return
```